src/matrix/matrixtdist.jl

```diff
@@ -0,0 +1,196 @@
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractMatrix,         Ω::AbstractPDMat)          = MatrixTDist(ν, M, PDMat(Σ), Ω)
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractMatrix,         Ω::LinearAlgebra.Cholesky) = MatrixTDist(ν, M, PDMat(Σ), PDMat(Ω))
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::LinearAlgebra.Cholesky, Ω::AbstractMatrix)         = MatrixTDist(ν, M, PDMat(Σ), PDMat(Ω))
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::LinearAlgebra.Cholesky, Ω::AbstractPDMat)          = MatrixTDist(ν, M, PDMat(Σ), Ω)
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::LinearAlgebra.Cholesky, Ω::LinearAlgebra.Cholesky) = MatrixTDist(ν, M, PDMat(Σ), PDMat(Ω))
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractPDMat,          Ω::AbstractMatrix)         = MatrixTDist(ν, M, Σ,        PDMat(Ω))
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractPDMat,          Ω::LinearAlgebra.Cholesky) = MatrixTDist(ν, M, Σ,        PDMat(Ω))
+
+#  -----------------------------------------------------------------------------
+#  REPL display
+#  -----------------------------------------------------------------------------
+
+ show(io::IO, d::MatrixTDist) = show_multline(io, d, [(:ν, d.ν), (:M, d.M), (:Σ, Matrix(d.Σ)), (:Ω, Matrix(d.Ω))])
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{MatrixTDist{T}}, d::MatrixTDist) where T <: Real
+    MM = convert(AbstractArray{T}, d.M)
+    ΣΣ = convert(AbstractArray{T}, d.Σ)
+    ΩΩ = convert(AbstractArray{T}, d.Ω)
+    MatrixTDist{T, typeof(MM), typeof(ΣΣ)}(T(d.ν), MM, ΣΣ, ΩΩ, T(d.logc0))
+end
+
+function convert(::Type{MatrixTDist{T}}, ν, M::AbstractMatrix, Σ::AbstractPDMat, Ω::AbstractPDMat, logc0) where T <: Real
+    MM = convert(AbstractArray{T}, M)
+    ΣΣ = convert(AbstractArray{T}, Σ)
+    ΩΩ = convert(AbstractArray{T}, Ω)
+    MatrixTDist{T, typeof(MM), typeof(ΣΣ)}(T(ν), MM, ΣΣ, ΩΩ, T(logc0))
+end
+
+#  -----------------------------------------------------------------------------
+#  Properties
+#  -----------------------------------------------------------------------------
+
+size(d::MatrixTDist) = size(d.M)
+
+rank(d::MatrixTDist) = minimum( size(d) )
+
+insupport(d::MatrixTDist, X::Matrix) = isreal(X) && size(X) == size(d)
+
+function mean(d::MatrixTDist)
+
+  n, p = size(d)
+
+  (d.ν + p - n > 1) ? (return d.M) : throw(ArgumentError("mean only defined for df + p - n > 1"))
+
+end
+
+mode(d::MatrixTDist) = d.M
+
+params(d::MatrixTDist) = (d.ν, d.M, d.Σ, d.Ω)
+
+@inline partype(d::MatrixTDist{T}) where {T <: Real} = T
+
+#  -----------------------------------------------------------------------------
+#  Evaluation
+#  -----------------------------------------------------------------------------
+
+function _matrixtdist_logc0(Σ::AbstractPDMat, Ω::AbstractPDMat, ν::Real)
```

Comment:
Underscore not needed here, it's fine to remove it

Reply:
Removed.